app/assets/main/react_components/User/CarouselCategoryButton.jsx

```diff
@@ -20,6 +30,28 @@ const CarouselCategoryButton = ({
     setAllCategories(false);
   };
 
+  const setCategory = (cat) => {
+    //***** HERE WE UPDATE THE CONTEXT */
```

Comment:
I see the reason why updateCategory wasnt named setCategory... Perhaps change them to the opposite?